lib/types/thermostat.js

```diff
@@ -0,0 +1,345 @@
+var inherits = require('util').inherits;
+
+module.exports = function (HAPnode, config, functions) {
+    var Accessory = HAPnode.Accessory;
+    var Service = HAPnode.Service;
+    var Characteristic = HAPnode.Characteristic;
+    var uuid = HAPnode.uuid;
+    var debug = HAPnode.debug;
+
+    var module = {};
+
+    module.newDevice = function (device, tempDisplayUnit) {
+        var temperatureDisplayUnit = tempDisplayUnit;
+        var Thermostat = {
+            device_url: function(request_type){
+                host = 'http://'+config.veraIP+':3480/data_request'
+                return host+'?id='+request_type+'&DeviceNum='+device.id
+            },
+
+            remoteRequest: function(url, params, callback){
+                debug("Requesting: %s", url);
+                //HAPnode.request.debug = true
+                return HAPnode.request({
+                    method:'GET',
+                    uri: url,
+                    qs: params,
+                    resolveWithFullResponse: true
+                }).then(callback.bind(this)).catch(function(e){
+                    debug(e.error);
+                    debug(e.options);
+                    debug(e.response);
+                });
+            },
+
+            getVeraVariable: function(serviceId, variable, callback){
+                var url = this.device_url('variableget');
+                var params = {serviceId: serviceId, Variable: variable}
+                return this.remoteRequest(url, params, callback);
+            },
+
+            executeVeraAction: function(params){
+                var url = this.device_url('action');
+                var callback = function(){};
+                return this.remoteRequest(url, params, callback);
+            },
+
+            getCurrentHeatingCoolingState: function(){
+                var serviceId = 'urn:upnp-org:serviceId:HVAC_UserOperatingMode1';
+                var variable = 'ModeStatus';
+                var callback = function (response){
+                    if (response.statusCode === 200){
+                        var vera_state = response.body;
+                        var state;
+                        switch(vera_state){
```

Comment:
@drewcovi well it all goes down to user mode vs operating(machine) mode. I actually thought of which one of this two to choose. I chose the user mode because that what you see as options (buttons etc.) in the interface and for me it made more sense to map interface options to homekit options.  To answer your question I don't know :) I only have heating so I can't see different options in action. Maybe someone else can pitch in?